dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1143,23 +1261,24 @@ CDefaultContextMenu::InvokeRegVerb(
 
     /* Get the browse flags to see if we need to browse */
     DWORD wFlags = BrowserFlagsFromVerb(lpcmi, pEntry);
-    BOOL bBrowsed = FALSE;
 
     for (i=0; i < m_cidl; i++)
     {
         /* Check if we need to browse */
-        if (wFlags > 0)
+        if (wFlags)
         {
-            /* In xp if we have browsed, we don't open any more folders.
-             * In win7 we browse to the first folder we find and
-             * open new windows for each of the rest of the folders */
-            if (bBrowsed)
-                continue;
-
             hr = TryToBrowse(lpcmi, m_apidl[i], wFlags);
             if (SUCCEEDED(hr))
             {
-                bBrowsed = TRUE;
+                /* In WinXP if we have browsed, we don't open any more folders.
+                 * In Win7 we browse to the first folder we find and
+                 * open new windows for each of the rest of the folders */
+                UINT ntver = RosGetProcessEffectiveVersion();
+                if (ntver >= _WIN32_WINNT_VISTA)
+                    wFlags = 0; // FIXME: = SBSP_NEWBROWSER | (wFlags & ~SBSP_SAMEBROWSER);
```

Comment:
This SBSP_.... flag is not supported yet?

Reply:
Correct. I don't think the current IShellBrowser cares about any of the SBSP_ flags.

It is no big deal for Explorer because it will just open the folder with ShellExecute instead when wFlags is 0. It is a nice thing to do for something like DirectoryOpus or anyone else that wants more control (if Explorer wants to implement tabs in the future etc)

Reply:
![image](https://github.com/reactos/reactos/assets/1969829/95d321db-4913-4432-b756-93047c45c797)
(Screenshot from Win2k3 PSDK)

It appears to be already supported in Win2k3 (or even WinXP?) at least, so maybe just set that flags combination? If our IShellBrowser ignores it, then no problem there won't be any behavioural change there.

Reply:
Because ROS Explorer does not handle the flags correctly.

I might look into that another time.

Reply:
ah ok, understandable then :+1: